annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -383,9 +383,8 @@ private Tree getNext(Tree actualNode, ASTPath astPath, int ix) {
                 } else if (astNode.childSelectorIs(ASTPath.BOUND)) {
                   return arg < 0 ? clazz.getExtendsClause()
                           : clazz.getImplementsClause().get(arg);
-                } else {
-                    return null;
                 }
+                return null;
```

Comment:
I propose to put this return statement between lines 382 and 383, which is the only place from which it can be executed.
This looks like a bug fix in AFU, which is great to have.